### ISSUE TITLE
Label with FontSize="Header" is enormous on Android and tiny on iOS

### ISSUE THREAD

**Issue by jlbilh** [dotnet/maui#7428]
New project.

Add this to the VerticalStackLayout:
```
            <Label
                FontSize="Header"
                HorizontalOptions="Center"
                Text="THE HEADER"/>
```

Windows:
![image](https://user-images.githubusercontent.com/102621630/169923454-0881cdf2-a4fa-4342-893e-3f8fe4530412.png)

Android 12:
![image](https://user-images.githubusercontent.com/102621630/169922952-ba8b0d16-4a71-4dc4-9dfe-3e1819159c76.png)

iOS 15:
![image](https://user-images.githubusercontent.com/102621630/169923417-c520db6f-4f23-46f3-a5db-d18da359ad13.png)

**Comment by beeradmoore** [comment]
Looks like the [underlying size](https://github.com/dotnet/maui/blob/81d1c0c25306ad756520c4829c345b6557d678ba/src/Controls/src/Core/Compatibility/Android/FontNamedSizeService.cs#L65) is using what Xamarin.Forms used to have until it was [fixed](https://github.com/xamarin/Xamarin.Forms/pull/12983).

**Comment by Dreamescaper** [comment]
Shouldn't ios be updated as well? It probably shouldn't be tiny as it is now.

**Comment by beeradmoore** [comment]
I don't think so. Looking at what is was [in Xamarin.Forms](https://github.com/xamarin/Xamarin.Forms/blob/5.0.0/Xamarin.Forms.Platform.iOS/Forms.cs#L348-L357) (for iOS) and was it is [in MAUI](https://github.com/dotnet/maui/blob/81d1c0c25306ad756520c4829c345b6557d678ba/src/Controls/src/Core/Compatibility/iOS/FontNamedSizeService.cs#L61-L70) (for iOS) it appears it is the same as before.

After some digging it looks like NamedSize [may be obsolete](https://github.com/dotnet/maui/pull/7433#issuecomment-1135763286) and on its way out and having it exist is just for compatibility sake? Unless its marked that way by accident.

**Comment by beeradmoore** [comment]
I also made a tool to try compare the font sizes, it.. somewhat works. It uses the named sizes from MAUI and compares them to the named sizes used on platform with their native names. Android doesn't work as I am unsure how to get font size from style type. I also didn't bother booting up my Windows PC and add compatibility for it.

[NamedFontTest.zip](https://github.com/dotnet/maui/files/8762322/NamedFontTest.zip)

(I don't know how to resize images in markup and place side by side so they are going to be big)

iOS Maui Size
![iOS Maui Size](https://user-images.githubusercontent.com/904737/170027339-a24c1422-f298-4572-824a-9baf25e26af8.png)


iOS Native Size
![iOS Native Size](https://user-images.githubusercontent.com/904737/170027437-c44e3063-6b0c-4d8c-a3a2-e1a6414c5214.png)

macOS Maui Size
![macOS Maui Size](https://user-images.githubusercontent.com/904737/170027498-4365f219-da46-4523-936a-ff2f8aad761d.png)

macOS Native Size
![macOS Native Size](https://user-images.githubusercontent.com/904737/170027535-ebc45b4e-3198-4661-9ed7-a7c6d9af658f.png)

Android Maui Size
![Android Maui Size](https://user-images.githubusercontent.com/904737/170027582-e67f9bc5-0dc3-4eac-9ec4-630ca9cb6a26.png)

Android Native Size
![Android Native Size](https://user-images.githubusercontent.com/904737/170027616-e509e50e-c2cb-486d-b36b-32eefc1ae7c6.png)
